resource/opentelemetry-resource-detector-container/src/opentelemetry/resource/detector/container/__init__.py

```diff
@@ -0,0 +1,93 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from opentelemetry.sdk.resources import Resource, ResourceDetector
+from opentelemetry.semconv.resource import ResourceAttributes
+
+logger = logging.getLogger(__name__)
+_DEFAULT_CGROUP_V1_PATH = "/proc/self/cgroup"
+_DEFAULT_CGROUP_V2_PATH = "/proc/self/mountinfo"
+_CONTAINER_ID_LENGTH = 64
+
+
+def _get_container_id_v1():
+    container_id = None
+    try:
+        with open(
+            _DEFAULT_CGROUP_V1_PATH, encoding="utf8"
+        ) as container_info_file:
+            for raw_line in container_info_file.readlines():
+                line = raw_line.strip()
+                if len(line) > _CONTAINER_ID_LENGTH:
+                    container_id = line[-_CONTAINER_ID_LENGTH:]
+                    break
+    except FileNotFoundError as exception:
+        logger.warning("Failed to get container id. Exception: %s", exception)
+    return container_id
+
+
+def _get_container_id_v2():
+    container_id = None
+    try:
+        with open(
+            _DEFAULT_CGROUP_V2_PATH, encoding="utf8"
+        ) as container_info_file:
+            for raw_line in container_info_file.readlines():
+                line = raw_line.strip()
+                if "containers" or "hostname" in line:
```

Comment:
This is an error (that is failing lint):

`"containers" or "hostname" in line` will always be `"containers"`. `"containers"` has an intrinsic boolean value of `true`, if there is a `true` value to the left of an `or` operator, then nothing at the right side of the `or` operator gets evaluated.

```suggestion
                if "containers" in line or "hostname" in line:
```